examples/diesel-mysql/main.rs

```diff
@@ -18,72 +18,55 @@ fn main() -> anyhow::Result<()> {
         "mysql://demo:password@localhost/jiff_diesel_example",
     )?;
 
-    example_datetime_decode(&mut conn)?;
-    example_datetime_encode(&mut conn)?;
+    example_datetime_roundtrip(&mut conn)?;
 
     Ok(())
 }
 
-/// Performs a decoding with all of Jiff's datetime types.
-fn example_datetime_decode(conn: &mut MysqlConnection) -> anyhow::Result<()> {
-    // Decoding jiff::Timestamp
-    let query = select(sql::<sql_types::Datetime>("'1970-01-01 00:00:00Z'"));
-    let got: jiff_diesel::Timestamp = query.get_result(conn)?;
-    let expected =
-        "1970-01-01T00:00:00Z".parse::<jiff::Timestamp>()?.to_diesel();
-    assert_eq!(expected, got);
+/// Performs a round-trip with all of Jiff's datetime types.
+fn example_datetime_roundtrip(
+    conn: &mut MysqlConnection,
+) -> anyhow::Result<()> {
+    diesel::table! {
+        jiffs {
+            id -> Integer, // Diesel tables require an ID column.
+            ts -> Datetime,
+            dt -> Timestamp,
+            d -> Date,
+            t -> Time
+        }
+    }
 
-    // Decoding jiff::civil::DateTime
-    let query = select(sql::<sql_types::Timestamp>("'2025-07-20 00:00:00'"));
-    let got: jiff_diesel::DateTime = query.get_result(conn)?;
-    let expected = civil::date(2025, 7, 20).at(0, 0, 0, 0).to_diesel();
-    assert_eq!(expected, got);
+    #[derive(Debug, PartialEq, QueryableByName)]
+    #[diesel(check_for_backend(diesel::mysql::Mysql))]
```

Comment:
What does this do? (I see it mentioned once in the crate docs of Diesel, but I still don't understand it.)

Reply:
This adds code which tests that types line up with the database schema and available `FromSql`/`ToSql` implementations, for the specified database backend.

If everything is good, nothing changes. If there was an error (which would later cause a compilation failure):

1. it exercises some of the type-driven machinery so that you get an error if your struct definition is incorrect, even before trying to use it in a query.
2. the error it generates tries to create an error span that points at the location where the mistake was made.
3. it can end up being more specific with what particular trait implementation is missing than you would get without it.

Reply:
So think of the naming as "check... [my code against trait implementations]... for backend... Mysql"

Reply:
Hmmm interesting. I guess that raises the next question, which is why is that opt-in? :-) But thank you for educating me!

Reply:
It's opt-in as it is something added after the last major release and the relevant code needs to know the backend type (`diesel::mysql::Mysql`) to perform this check. There is unfortunately no way to get that type for old impls without the attribute without breaking stuff as we cannot assume that users only enabled a single backend.